### PR TITLE
add codeceptjs with tests

### PR DESCRIPTION
**Pull request by patrjam**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -26,7 +30,10 @@
     "test": "react-scripts test",
     "eject": "react-scripts eject",
     "predeploy": "npm run build",
-    "deploy": "gh-pages -d build"
+    "deploy": "gh-pages -d build",
+    "test:lvl-1": "codeceptjs run --config=./automation-example/configs/level-1.config.conf.js --debug",
```
Comment (Trawais) [inline package.json]: Do we have some better name for these "levels"?

File: .gitignore [hunk 2]
```diff
@@ -24,3 +24,10 @@ yarn-error.log*
 
 #build
 /build
+
+#output for automation tests
+/output
+
+.npmrc
+
+.env
```
Comment (Trawais) [inline .gitignore]: missing an empty line at the end of the file

File: automation-example/configs/level-1.config.conf.js [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+  tests: '../tests/level-1/*_test.js',
+  output: '../../output',
+  helpers: {
+    WebDriver: {
+      url: 'http://localhost:3000',
+      browser: 'chrome'
+    }
+  },
+  include: {
+    I: '../../steps_file.js'
```
Comment (CDJojn) [inline automation-example/configs/level-1.config.conf.js]: why are you not use jsqa/step_file.js ??

File: automation-example/configs/level-1.config.conf.js [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+  },
+  bootstrap: null,
+  mocha: {},
+  name: 'react-web-example',
+  plugins: {
```
Comment (CDJojn) [inline automation-example/configs/level-1.config.conf.js]: no wdio plugin?

File: automation-example/pages/App/App.js [hunk 5]
```diff
@@ -0,0 +1,3 @@
+module.exports = {
+  application: '[id="root"]',
```
Comment (CDJojn) [inline automation-example/pages/App/App.js]: why is here this file?

File: automation-example/pages/Messages/Success.js [hunk 6]
```diff
@@ -0,0 +1,4 @@
+module.exports = {
```
Comment (CDJojn) [inline automation-example/pages/Messages/Success.js]: I'm not sure if this is good approach... this is not PAGE, just text message and why is this file not part of autoload or part of page where is located.... missing context.

File: automation-example/pages/Header/Header.js [hunk 7]
```diff
@@ -0,0 +1,3 @@
+module.exports = {
```
Comment (CDJojn) [inline automation-example/pages/Header/Header.js]: same situation this should be components or what?

File: automation-example/tests/level-3/fill-form-3_test.js [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+});
+
+Scenario("Fill mandatory fields in form - @customLocator(data-qa)", (I) => {
+  I.amOnPage("/");
+
+  I.fillField(form.inputs.name, faker.name.firstName());
+  form.fillInRandomEmail(faker.internet.email());
+  I.fillField(form.inputs.comment, faker.lorem.text());
+
+  //NOTE: need to add data-qa in to code
```
Comment (CDJojn) [inline automation-example/tests/level-3/fill-form-3_test.js]: clear note?

File: steps_file.js [hunk 9]
```diff
@@ -0,0 +1,10 @@
+// in this file you can append custom step methods to 'I' object
+
+module.exports = function() {
```
Comment (CDJojn) [inline steps_file.js:3]: remove and use from JSQA

**Review by CDJojn — CHANGES_REQUESTED** [review]

File: automation-example/pages/Form/FormPage.js [hunk 10]
```diff
@@ -0,0 +1,62 @@
...
+    I.fillField(this.inputs.name, name);
+    I.fillField(this.inputs.email, email);
+    I.fillField(this.inputs.comment, comment);
+    this.submitForm();
+  },
+
+  /*
+example for data-client
+*/
+  async fillInNameWithScudUser() {
```
Comment (CDJojn) [inline automation-example/pages/Form/FormPage.js]: split into 2 methods - fillwithName() and getScudUser() - what is direct method of DataClientHelper - this call should be part of test, not part od PageObject

**Pull request merged**